met-web/src/components/userManagement/listing/ActionsDropDown.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useMemo, useContext } from 'react';
+import { MenuItem, Select } from '@mui/material';
+import { User } from 'models/user';
+import { Palette } from 'styles/Theme';
+import { AddUserModal } from './AddUserModal';
+import { AssignRoleModal } from './AssignRoleModal';
+import { UserManagementContext } from './UserManagementContext';
+import { USER_GROUP } from 'models/user';
+
+interface ActionDropDownItem {
+    value: number;
+    label: string;
+    action?: () => void;
+    condition?: boolean;
+}
+export const ActionsDropDown = ({ selectedUser }: { selectedUser: User }) => {
+    const { addUserModalOpen, setAddUserModalOpen, assignRoleModalOpen, setassignRoleModalOpen, setUser } =
+        useContext(UserManagementContext);
+
+    const handleChange = (selectedItem: number) => {
+        if (selectedItem == 1) {
+            setassignRoleModalOpen(true);
+        } else if (selectedItem == 2) {
+            setAddUserModalOpen(true);
+        }
+    };
+
+    const hasNoRole = (): boolean => {
+        if (selectedUser.main_group) {
+            return false;
+        }
+        return true;
+    };
+
+    const isAdmin = (): boolean => {
+        if (selectedUser?.main_group == USER_GROUP.ADMIN.label) {
+            return true;
+        }
+        return false;
+    };
+
+    const ITEMS: ActionDropDownItem[] = useMemo(
+        () => [
+            {
+                value: 1,
+                label: 'Assign Role',
+                action: () => {
+                    {
+                        setUser(selectedUser);
+                        assignRoleModalOpen ? <AssignRoleModal /> : null;
```

Comment:
why do we need handleChange. Can we not just assign the appropriate modal open here? Also we are mixing javascript and JSX here. I think this  action should just perform an action and not add any jsx

Reply:
removed handleChange and made changes to assign the appropriate modal open, thanks